openapi.yaml

```diff
@@ -0,0 +1,484 @@
+    Cluster:
+      allOf:
+        - $ref: '#/components/schemas/ClusterId'
+        - $ref: '#/components/schemas/ClusterProperties'
+        - $ref: '#/components/schemas/ClusterRequiredProperties'
+    Reason:
+      type: object
+      description: A reason for responses
+      properties:
+        reason:
+          type: string
+          description: The reason message
+          example: Because of reasons
+    Inventory:
+      type: object
+      description: Inventory data of a cluster
+      required:
+      - cluster
+      properties:
+        cluster:
+          type: string
+        inventory:
+          type: object
+          additionalProperties: {}
+  requestBodies:
+    CreateTenant:
+      description: Create a new tenant
+      content:
+        application/json:
+          schema:
+            allOf:
+            - $ref: '#/components/schemas/TenantProperties'
+            - $ref: '#/components/schemas/TenantRequiredProperties'
+    CreateCluster:
+      description: Create a new Cluster
+      content:
+        application/json:
+          schema:
+            allOf:
+            - $ref: '#/components/schemas/ClusterProperties'
+            - $ref: '#/components/schemas/ClusterRequiredProperties'
+    StoreInventory:
+      description: Inventory data of a cluster
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Inventory'
+  parameters:
+    TenantId:
+      name: tenantId
+      in: path
+      required: true
+      description: Distinct id of the tenant.
+      schema:
+        type: string
+        minimum: 1
+    ClusterId:
+      name: clusterId
+      in: path
+      required: true
+      description: Distinct id of the cluster.
+      schema:
+        type: string
+        minimum: 1
+tags:
+  - name: tenant
+    description: Management of tenants
+  - name: cluster
+    description: Management of clusters
+  - name: bootstrapping
+    description: Cluster bootstrapping
+  - name: inventory
+    description: Cluster inventory time based data
+  - name: system
+    description: API system
+paths:
+  /tenants:
+    get:
+      summary: Returns a list of tenants.
+      description: List of all tenants available in the API
+      tags:
+        - tenant
+      responses:
+        '200':
+          description: Tenants found.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Tenant'
+                example:
+                - id: aezoo6
+                  name: bigcorp
+                  displayName: Big Corp.
+                  gitRepo: https://github.com/bigcorp/commodore-config.git
+                - id: os3ce3
+                  name: acmecorp
+                  displayName: Acme Corp. (Subtenant of Big Corp)
+                  tenant: aezoo6
+                  gitRepo: https://github.com/acmecorp/commodore-config.git
+        '404':
```

Comment:
This should probably also return a reason to keep error handling consistent? Maybe we could even introduce an error schema to return on all errors which includes the reason?

Reply:
My idea was to just have the reason returned when the HTTP status is ambiguous, f.e. `400` just says "Bad Request" which can be of many reasons, whereas `404` clearly says `Not Found` leaving little ambiguation. Do you think it's worth to send a reason to every HTTP return status?

Reply:
It might help when debugging, as an empty response with just the status code usually confuses me :smile: (which might very well be personal preference, so feel free to ignore).

Reply:
/ignore :smile: -> https://httpie.org/